src/com/trilead/ssh2/crypto/dh/DhExchange.java

```diff
@@ -1,160 +1,212 @@
-
+/**
+ *
+ */
 package com.trilead.ssh2.crypto.dh;
 
-import java.io.UnsupportedEncodingException;
+import javax.crypto.KeyAgreement;
+import javax.crypto.interfaces.DHPrivateKey;
+import javax.crypto.interfaces.DHPublicKey;
+import javax.crypto.spec.DHParameterSpec;
+import javax.crypto.spec.DHPublicKeySpec;
+import java.io.IOException;
 import java.math.BigInteger;
-import java.security.SecureRandom;
-
-import com.trilead.ssh2.crypto.digest.HashForSSH2Types;
-import com.trilead.ssh2.log.Logger;
-
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
 
 /**
- * DhExchange.
- * 
- * @author Christian Plattner, plattner@trilead.com
- * @version $Id: DhExchange.java,v 1.2 2008/04/01 12:38:09 cplattne Exp $
+ * @author kenny
```

Comment:
we should keep the previous credits

```suggestion
 * DhExchange.
 *
 * @author Christian Plattner, plattner@trilead.com, kenny
 * @version $Id: DhExchange.java,v 1.2 2008/04/01 12:38:09 cplattne Exp $
```

Reply:
Done!